app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js

```diff
@@ -9,29 +9,41 @@ var initializer = function() {
     var url = boolSwitch.data('url');
     var value = boolSwitch.data('value');
     var successMessage = boolSwitch.data('success_message');
+    var confirmToggleTrueMessage = boolSwitch.data('confirm_toggle_true_message');
+    var confirmToggleFalseMessage = boolSwitch.data('confirm_toggle_false_message');
     var data = { id: objectId };
     data[model] = {};
     data[model][field] = !value;
 
-    $.ajax({
-      url: url,
-      data: data,
-      dataType: 'json',
-      headers : {'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')},
-      error: function() {
-        var errorMsg = 'Error: Update Unsuccessful';
-        alert(errorMsg);
-      },
-      success: function() {
-        boolSwitch.data('value', !value);
-        boolSwitch.toggleClass('on');
-        if (!boolSwitch.hasClass('notify-success')) return;
-        $(function() {
-          setTimeout(alert(successMessage), 500);
-        });
-      },
-      type: 'PATCH',
-    });
+    var resp = true
+
+    if (value && confirmToggleFalseMessage) {
+      resp = confirm(confirmToggleFalseMessage);
+    } else if (!value && confirmToggleTrueMessage){
+      resp = confirm(confirmToggleTrueMessage);
+    }
+
+    if (resp) {
+      $.ajax({
+        url: url,
+        data: data,
+        dataType: 'json',
+        headers : {'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')},
```

Comment:
Extra space after key 'headers'                  key-spacing<br>A space is required after '{'                    object-curly-spacing<br>A space is required before '}'                   object-curly-spacing